src/metadata/metadata.controller.spec.ts

```diff
@@ -0,0 +1,169 @@
+import { DeepMocked } from '@golevelup/ts-jest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { BigNumber } from '@polymeshassociation/polymesh-sdk';
+import {
+  MetadataEntry,
+  MetadataLockStatus,
+  MetadataType,
+  TxTags,
+} from '@polymeshassociation/polymesh-sdk/types';
+import { when } from 'jest-when';
+
+import { TransactionType } from '~/common/types';
+import { CreateMetadataDto } from '~/metadata/dto/create-metadata.dto';
+import { SetMetadataDto } from '~/metadata/dto/set-metadata.dto';
+import { MetadataController } from '~/metadata/metadata.controller';
+import { MetadataService } from '~/metadata/metadata.service';
+import { CreatedMetadataEntryModel } from '~/metadata/models/created-metadata-entry.model';
+import { MetadataDetailsModel } from '~/metadata/models/metadata-details.model';
+import { MetadataEntryModel } from '~/metadata/models/metadata-entry.model';
+import { MetadataValueModel } from '~/metadata/models/metadata-value.model';
+import { createMockTransactionResult, MockMetadataEntry } from '~/test-utils/mocks';
+import { mockMetadataServiceProvider } from '~/test-utils/service-mocks';
+
+describe('MetadataController', () => {
+  let controller: MetadataController;
+  let mockService: DeepMocked<MetadataService>;
+  let ticker: string;
+  let type: MetadataType;
+  let id: BigNumber;
+
+  beforeEach(async () => {
+    ticker = 'TICKER';
+    type = MetadataType.Local;
+    id = new BigNumber(1);
+
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [MetadataController],
+      providers: [mockMetadataServiceProvider],
+    }).compile();
+
+    mockService = mockMetadataServiceProvider.useValue as DeepMocked<MetadataService>;
+
+    controller = module.get<MetadataController>(MetadataController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  describe('getMetadata', () => {
+    it('should return the list of all metadata for a given ticker', async () => {
+      const mockMetadataEntry = new MockMetadataEntry();
+      when(mockService.findAll)
+        .calledWith(ticker)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .mockResolvedValue([mockMetadataEntry as any]);
+
+      const result = await controller.getMetadata({ ticker });
+
+      expect(result).toEqual({
+        results: [new MetadataEntryModel({ asset: ticker, type, id })],
+      });
+    });
+  });
+
+  describe('getSingleMetadata', () => {
+    it('should return the Metadata details for a specific type and ID', async () => {
+      const mockMetadataEntry = new MockMetadataEntry();
```

Comment:
- use `createMock`, specify the assertion values
- remove `as any` cast

`mockResolvedValue` should be able to stay as they are